google/cloud/datastore/_http.py

```diff
@@ -464,6 +488,19 @@ def reserve_ids(self, request, retry=None, timeout=None):
             self.client._client_info,
             request_pb,
             _datastore_pb2.ReserveIdsResponse,
+            database_id,
             retry=retry,
             timeout=timeout,
         )
+
+
+def _update_headers(headers, project_id, database_id=None):
+    """Update the request headers.
+    Pass the project id, or optionally the database_id if provided.
+    """
+    if database_id:
```

Comment:
nit - it would be cleaner to keep the setting of the project id outside of the if else, and then only append the database id if it is present.

Reply:
fixed